tests/lib/fakeinstaller/main.go

```diff
@@ -0,0 +1,285 @@
+// an slice with the paths where the filesystems have been mounted to.
+func createAndMountFilesystems(bootDevice string, volumes map[string]*gadget.Volume) ([]string, error) {
+	// only support a single volume for now
+	if len(volumes) != 1 {
+		return nil, fmt.Errorf("got unexpected number of volumes %v", len(volumes))
+	}
+
+	disk, err := disks.DiskFromDeviceName(bootDevice)
+	if err != nil {
+		return nil, err
+	}
+	vol := firstVol(volumes)
+
+	var mountPoints []string
+	for _, stru := range vol.Structure {
+		if stru.Label == "" || stru.Filesystem == "" {
+			continue
+		}
+
+		part, err := disk.FindMatchingPartitionWithPartLabel(stru.Label)
+		if err != nil {
+			return nil, err
+		}
+		// XXX: reuse
+		// gadget/install/content.go:mountFilesystem() instead
+		// (it will also call udevadm)
+		if err := mkfs.Make(stru.Filesystem, part.KernelDeviceNode, stru.Label, 0, 0); err != nil {
+			return nil, err
+		}
+
+		// mount
+		mountPoint := runMntFor(stru.Label)
+		if err := os.MkdirAll(mountPoint, 0755); err != nil {
+			return nil, err
+		}
+		// XXX: is there a better way?
+		if output, err := exec.Command("mount", part.KernelDeviceNode, mountPoint).CombinedOutput(); err != nil {
+			return nil, osutil.OutputErr(output, err)
+		}
+		mountPoints = append(mountPoints, mountPoint)
+	}
+
+	return mountPoints, nil
+}
+
+func unmountFilesystems(mntPts []string) error {
+	for _, mntPt := range mntPts {
+		if output, err := exec.Command("umount", mntPt).CombinedOutput(); err != nil {
+			return osutil.OutputErr(output, err)
+		}
+	}
+	return nil
+}
+
+func createClassicRootfsIfNeeded(rootfsCreator string) error {
+	dst := runMntFor("ubuntu-data")
+
+	if output, err := exec.Command(rootfsCreator, dst).CombinedOutput(); err != nil {
+		return osutil.OutputErr(output, err)
+	}
+
+	return nil
+}
+
+func createSeedOnTarget(bootDevice, seedLabel string) error {
+	// XXX: too naive?
+	dataMnt := runMntFor("ubuntu-data")
+	src := dirs.SnapSeedDir
+	dst := dirs.SnapSeedDirUnder(dataMnt)
+	if err := os.MkdirAll(filepath.Dir(dst), 0755); err != nil {
+		return err
+	}
+	if output, err := exec.Command("cp", "-a", src, dst).CombinedOutput(); err != nil {
+		return osutil.OutputErr(output, err)
+	}
+
+	return nil
+}
+
+func run(seedLabel, bootDevice, rootfsCreator string) error {
+	if len(os.Args) != 4 {
+		// xxx: allow installing real UC without a classic-rootfs later
+		return fmt.Errorf("need seed-label, target-device and classic-rootfs as argument\n")
+	}
+	os.Setenv("SNAPD_DEBUG", "1")
+	logger.SimpleSetup()
+
+	logger.Noticef("installing on %q", bootDevice)
+
+	cli := client.New(nil)
+	details, err := cli.SystemDetails(seedLabel)
+	if err != nil {
+		return err
+	}
+	// TODO: grow the data-partition based on disk size
+	laidoutStructs, err := createPartitions(bootDevice, details.Volumes)
+	if err != nil {
+		return fmt.Errorf("cannot setup partitions: %v", err)
+	}
+	fmt.Println("laidoutStructs len:", len(laidoutStructs))
```

Comment:
Should this be a `logger.Debugf` instead?

Reply:
Right, changed now